test/ibmq/test_ibmq_job.py

```diff
@@ -290,6 +291,13 @@ def test_retrieve_job_uses_appropriate_backend(self, backend, provider):
                               backend_2.retrieve_job, job_1.job_id())
         self.assertIn('belongs to', str(context_manager.warning))
 
+        # Cleanup
+        try:
+            job_1.cancel()
+            job_2.cancel()
```

Comment:
Good point! Yes it's certainly worth separating them since this test case uses a real device and therefore any leftover jobs would consume credits, which may cause other tests to fail.